sdk/src/main/java/co/omise/android/ui/AuthorizingPaymentActivity.kt

```diff
@@ -185,6 +172,25 @@ class AuthorizingPaymentActivity : AppCompatActivity() {
         }
     }
 
+    private fun setupActionBarTitle() {
+        supportActionBar?.title = uiCustomization.uiCustomization.toolbarCustomization?.headerText
+            ?: getString(R.string.title_authorizing_payment)
+    }
+
+    private fun handlePaymentAuthorization() {
+        val authUrlString = verifier.authorizedURLString
+        val authUrl=verifier.authorizedURL
+        // check for legacy payments that require web view
+        if (authUrlString.endsWith("/pay")) {
+            setupWebView()
+        } else {
+            // Check if the URL needs to be opened externally
+            if (verifier.verifyExternalURL(authUrl)) {
+                openDeepLink(authUrl)
+            }
+            observeData()
+        }
```

Comment:
```suggestion
        } else if (verifier.verifyExternalURL(authUrl)) {
            openDeepLink(authUrl)
        }

        observeData()
```

Reply:
@aashishgurung  `observeData()` should not be launched in case of legacy payment

Reply:
How does our current logic handle that?

- if URL ends with `/pay`, setup web view
- if not, it will go to `else` block and opens deep link if authURL is verified as externalUrl.
- if not it will trigger observeData.

So, `observeData` will still be triggered in our current implementation.

Reply:
yes it should be triggered on all cases except legacy because previously `observe` was called on all cases so now it should be called on all except legacy

Reply:
Got it. Thanks for explaining